github/resource_github_repository_collaborator.go

```diff
@@ -158,6 +172,10 @@ func resourceGithubRepositoryCollaboratorRead(d *schema.ResourceData, meta inter
 	return nil
 }
 
+func resourceGithubRepositoryCollaboratorUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Adding an update function was required if we are to add the `permission_diff_suppression` option without being `ForceNew`.  This enables a smoother upgrade option as users can turn on this workaround without triggering a recreate.  Downside is we'll have to keep this no-op update in mind if we ever want to update actual state.